### DISCUSSION TITLE
Deceptive wording in Recover lost objects "recover selected objects" deletes the object?

### DISCUSSION THREAD

**Discussion by Spharad** [gitextensions/gitextensions#11248] [Q&A]
### Environment

- GitExtensions version:3.5.4.12724
- GIT version: 2.39.1.windows.1
- OS version: Windows 10
- .NET version: irrelevant

### Issue description

I "accidentally" reset changes to a staged file.  When I say accidentally, I thought I was typing in the comments section, and I typed r e which reset the current file.  Then I tried to recover it with the Repository/Maintenance/recover lost objects... menu.  I could see my lost module as a blob...wonderful.  Then I selected the blob and clicked the button "Recover selected objects" and it disappeared, the object did NOT appear in my directory and now apparently it's truly lost.

THE MAIN PROBLEM is the deceptive wording of the button "Recover selected objects".  The apparent meaning is that it should recover your files.  In reality, it completely removes your files.  The wording needs to be changed.

ANOTHER problem on the UI involves the key combination "R-E", which resets changes.  The problem is that if you THINK you are in the comments section and start typing "re" it will reset the first file (which is selected by default).  This is just as bad as having no confirmation at all.  And "re" is a pretty common prefix!
![image](https://github.com/gitextensions/gitextensions/assets/134089275/33cde6e5-317c-4bfb-9f19-d98303e4e0f7)
My recommendation is to remove the shortcut "E" from that dialog.  Escape is fine since that isn't something you would normally type.

### Steps to reproduce

Follow the steps above

### Did this work in previous version of GitExtensions?

don't know

**Comment by gerhardol** [comment]
Staged files are never stored as objects in Git. So if you reset, there is no "backup".

**Comment by pmiossec** [comment]
@gerhardol git BLOB objects are created when you stage them (so you can recover them) whereas TREE objects are created when you commit. See (the 2nd answer will surprise you 🤣  ):
https://stackoverflow.com/questions/11094968/in-git-how-can-i-recover-a-staged-file-that-was-reverted-prior-to-committing/58853981

@Spharad The recover UI is far from being perfect because the UI allow to handle the recovery commits and blob objects that 2 things totally different.
You used a button which is more to recover commits than blobs.
As a consequence the command has created a TAG pointing directly toward a BLOB (whereas normally a TAG should point toward a COMMIT object). And because of that, it's no more a dangling blob and so is removed from the list (but is still available in git object "database")

Side note: I don't have a clue why the tag created doesn't appear inside GitExtensions tag list. Maybe because it is not pointing toward a commit but a blob (and so is rejected by a regex? Not spent time investigated/ing further.).

To recover the file content, do that...

First find the tag name of the created tag:

    git tag | grep LOST_FOUND

The tag should have the name `LOST_FOUND_1` but the command let you be sure...

Then you have 2 possibilities:

1. (easier) recover the content to a file in the working directory with the command (update tag and file names)

    git cat-file -p LOST_FOUND_1 > my_recovered_file.txt

Or:
2. Delete the tag

     git tag -d LOST_FOUND_1

 and it will re-appear in GitExtensions recover UI and you will be able to restore it with the "Save as...." feature:

![image](https://github.com/gitextensions/gitextensions/assets/460196/11e62fe2-f809-4b9d-a663-8fd4c441d4ac)


Once you recovered your file, delete the tag anyway...

**Reply by Spharad** [reply]
![image](https://github.com/gitextensions/gitextensions/assets/134089275/719ed9e6-416d-4acb-85b2-4e7fd10a86d0)
 I updated the original post.  The problem is the wording of the button Recover selected objects, which sound like it should recover the files.  This is terrible wording.  I don't know if there are other use cases for the GitExtensions option "Recover lost objects" but it was certainly not what I expected.

**Comment by Spharad** [comment]
@pmiossec @gerhardol   Now that I have updated the description, can this post be reopened as a bug?
